packages/business-rules/src/schemas/full-appeal/update.js

```diff
@@ -18,7 +20,20 @@ const update = pinsYup
     state: pinsYup.string().oneOf(Object.values(APPEAL_STATE)).required(),
     appealType: pinsYup.string().oneOf(Object.values(APPEAL_ID)).required(),
     decisionDate: pinsYup.lazy((decisionDate) => {
-      return pinsYup.date().isInThePast(decisionDate).transform(parseDateString).required();
+      return pinsYup
+        .date()
+        .isInThePast(decisionDate)
+        .test('decisionDate', 'decisionDate is not within dealine period', function test() {
+          return (
+            appeal.decisionDate.isWithinDecisionDateExpiryPeriod(
+              decisionDate,
+              this.options.parent.appealType,
+              this.options.parent.eligibility.applicationDecision,
+            ) || createYupError.call(this, 'must be before the deadline date')
+          );
+        })
+        .transform(parseDateString)
+        .required();
```

Comment:
It would make this validator more reuseable if it was created as a method of `pinsYup`

Also, this validator should be used in the `insert` schema too so that it does the same check when the user submits the application

Reply:
Reinstated
Added the validator to `Insert` schema as well

Reply:
Ok fair enough